The intent here is to permit compilation under C++20 with an option to support then-included `fmt`.  That is good.  But we can test that locally:

```diff
modified   src/Makevars
@@ -1,3 +1,5 @@
+CXX_STD = CXX20
+PKG_CXXFLAGS = -DSPDLOG_USE_STD_FORMAT
 
 ## We need the spdlog headers
 PKG_CPPFLAGS =	-I../inst/include/

```

When I do that, I get both C++20 and the new define.  And then the build fails:

```
formatter.cpp: In instantiation of ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&, std::index_sequence<S ...>) [with long unsigned int ...S = {0}; std::format_args = std:
:basic_format_args<std::basic_format_context<std::__format::_Sink_iter<char>, char> >; std::index_sequence<S ...> = std::integer_sequence<long unsigned int, 0>]’:                                                 
formatter.cpp:33:25:   required from ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&) [with long unsigned int size = 1; std::format_args = std::basic_format_args<std::basi
c_format_context<std::__format::_Sink_iter<char>, char> >]’                                                                                                                                                        
formatter.cpp:40:74:   required from here                                                                                                                                                                          
formatter.cpp:28:33: warning: returning reference to temporary [-Wreturn-local-addr]                     
   28 |     return std::make_format_args(vec[S]...);                                                     
      |            ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~                                                                                                                                                                
formatter.cpp: In instantiation of ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&, std::index_sequence<S ...>) [with long unsigned int ...S = {0, 1}; std::format_args = s
td::basic_format_args<std::basic_format_context<std::__format::_Sink_iter<char>, char> >; std::index_sequence<S ...> = std::integer_sequence<long unsigned int, 0, 1>]’:
formatter.cpp:33:25:   required from ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&) [with long unsigned int size = 2; std::format_args = std::basic_format_args<std::basi
c_format_context<std::__format::_Sink_iter<char>, char> >]’    
formatter.cpp:41:74:   required from here                                                                
formatter.cpp:28:33: warning: returning reference to temporary [-Wreturn-local-addr]             
formatter.cpp: In instantiation of ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&, std::index_sequence<S ...>) [with long unsigned int ...S = {0, 1, 2}; std::format_args 
= std::basic_format_args<std::basic_format_context<std::__format::_Sink_iter<char>, char> >; std::index_sequence<S ...> = std::integer_sequence<long unsigned int, 0, 1, 2>]’:                                     
formatter.cpp:33:25:   required from ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&) [with long unsigned int size = 3; std::format_args = std::basic_format_args<std::basi
c_format_context<std::__format::_Sink_iter<char>, char> >]’                                                                                                                                                        
formatter.cpp:42:74:   required from here                                                                
formatter.cpp:28:33: warning: returning reference to temporary [-Wreturn-local-addr]
formatter.cpp: In instantiation of ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&, std::index_sequence<S ...>) [with long unsigned int ...S = {0, 1, 2, 3}; std::format_ar
gs = std::basic_format_args<std::basic_format_context<std::__format::_Sink_iter<char>, char> >; std::index_sequence<S ...> = std::integer_sequence<long unsigned int, 0, 1, 2, 3>]’:
formatter.cpp:33:25:   required from ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&) [with long unsigned int size = 4; std::format_args = std::basic_format_args<std::basi
c_format_context<std::__format::_Sink_iter<char>, char> >]’
formatter.cpp:43:74:   required from here
formatter.cpp:28:33: warning: returning reference to temporary [-Wreturn-local-addr]
formatter.cpp: In instantiation of ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&, std::index_sequence<S ...>) [with long unsigned int ...S = {0, 1, 2, 3, 4}; std::format
_args = std::basic_format_args<std::basic_format_context<std::__format::_Sink_iter<char>, char> >; std::index_sequence<S ...> = std::integer_sequence<long unsigned int, 0, 1, 2, 3, 4>]’:
formatter.cpp:33:25:   required from ‘std::format_args&& unpack_vector(const std::vector<std::__cxx11::basic_string<char> >&) [with long unsigned int size = 5; std::format_args = std::basic_format_args<std::basi
c_format_context<std::__format::_Sink_iter<char>, char> >]’
formatter.cpp:44:74:   required from here
formatter.cpp:28:33: warning: returning reference to temporary [-Wreturn-local-addr]

[.... many more lines omitted ...]

```

It also fails to compile RcppExports.cpp, a generated file (using Rcpp::compileAttributes()).  I tried g++-13, g++-14, clang++-17 and clang++-18.  All failed.

--- a/.gitignore
+++ b/.gitignore
@@ -1,6 +1,9 @@
 # Retrieved 2017-Oct-12 from https://github.com/github/gitignore/blob/master/R.gitignore
 # Licensed under CC0-1.0 https://github.com/github/gitignore/blob/master/LICENSE
 
+# VSCode
+.vscode 
+
 # History files
 .Rhistory
 .Rapp.history

--- a/.gitignore
+++ b/.gitignore
@@ -1,9 +1,6 @@
 # Retrieved 2017-Oct-12 from https://github.com/github/gitignore/blob/master/R.gitignore
 # Licensed under CC0-1.0 https://github.com/github/gitignore/blob/master/LICENSE
 
-# VSCode
-.vscode 
-
 # History files
 .Rhistory
 .Rapp.history
@@ -44,3 +41,6 @@ docs/mkdmt-src/docs
 
 # Emacs
 *~
+
+# VSCode
+.vscode 

--- a/inst/include/rcpp_sink.h
+++ b/inst/include/rcpp_sink.h
@@ -30,6 +30,7 @@ protected:
         spdlog::memory_buf_t formatted;
         spdlog::sinks::base_sink<Mutex>::formatter_->format(msg, formatted);
 
+        // When using C++20 with SPDLOG_USE_STD_FORMAT flag spdlog::memory_buf_t is an alias of std::string
         #ifdef SPDLOG_USE_STD_FORMAT
         Rcpp::Rcout << formatted;
         #else

--- a/inst/include/rcpp_sink.h
+++ b/inst/include/rcpp_sink.h
@@ -29,7 +29,12 @@ protected:
         // sending it to its final destination:
         spdlog::memory_buf_t formatted;
         spdlog::sinks::base_sink<Mutex>::formatter_->format(msg, formatted);
+
+        #ifdef SPDLOG_USE_STD_FORMAT
+        Rcpp::Rcout << formatted;
+        #else
         Rcpp::Rcout << fmt::to_string(formatted);
+        #endif
     }
 
     void flush_() override {

--- a/inst/include/spdl.h
+++ b/inst/include/spdl.h
@@ -7,6 +7,10 @@
 // expose fmt::format via the courses in spdlog
 #include <spdlog/fmt/fmt.h>
 
+#ifdef SPDLOG_USE_STD_FORMAT
+#include <string_view>
+#endif
+
 // for convenience define cuter ones in another (shorter) namespace
 namespace spdl {
     inline void setup(const std::string& name = "default",
@@ -30,6 +34,26 @@ namespace spdl {
     // and many packages have opted into C++11 (or newer) but the check does not hurt
     #if __cplusplus >= 201103L
 
+    #ifdef SPDLOG_USE_STD_FORMAT 
+    template <typename... Args>
+    inline void trace(const char* fmt, Args&&... args ) { RcppSpdlog::log_trace(std::vformat(std::string_view(fmt), std::make_format_args(args...)).c_str()); }
+
+    template <typename... Args>
+    inline void debug(const char* fmt, Args&&... args ) { RcppSpdlog::log_debug(std::vformat(std::string_view(fmt), std::make_format_args(args...)).c_str()); }
+
+    template <typename... Args>
+    inline void info(const char* fmt, Args&&... args ) { RcppSpdlog::log_info(std::vformat(std::string_view(fmt), std::make_format_args(args...)).c_str()); }
+
+    template <typename... Args>
+    inline void warn(const char* fmt, Args&&... args ) { RcppSpdlog::log_warn(std::vformat(std::string_view(fmt), std::make_format_args(args...)).c_str()); }
+
+    template <typename... Args>
+    inline void error(const char* fmt, Args&&... args ) { RcppSpdlog::log_error(std::vformat(std::string_view(fmt), std::make_format_args(args...)).c_str()); }
+
+    template <typename... Args>
+    inline void critical(const char* fmt, Args&&... args ) { RcppSpdlog::log_critical(std::vformat(std::string_view(fmt), std::make_format_args(args...)).c_str()); }
+    
+    #else
     template <typename... Args>
     inline void trace(const char* fmt, Args&&... args ) { RcppSpdlog::log_trace(fmt::format(fmt, std::forward<Args>(args)... ).c_str()); }
 
@@ -47,7 +71,8 @@ namespace spdl {
 
     template <typename... Args>
     inline void critical(const char* fmt, Args&&... args ) { RcppSpdlog::log_critical(fmt::format(fmt, std::forward<Args>(args)... ).c_str()); }
-
+    #endif
+    
     #endif // if C++11
 
     inline Rcpp::XPtr<spdlog::stopwatch> stopwatch() { return RcppSpdlog::get_stopwatch(); }

--- a/src/formatter.cpp
+++ b/src/formatter.cpp
@@ -3,19 +3,21 @@
 
 #include <RcppSpdlog>
 
-#ifdef SPDLOG_USE_STD_FORMAT
+#if defined(SPDLOG_USE_STD_FORMAT) && __cplusplus >= 202002L
+#define NAMESPACE std
+#else
+#define NAMESPACE fmt
+#endif
 
 template<std::size_t... S>
-std::format_args&& unpack_vector(const std::vector<std::string>& vec, std::index_sequence<S...>) {
-    return std::make_format_args(vec[S]...);
+NAMESPACE::format_args unpack_vector(const std::vector<std::string>& vec, std::index_sequence<S...>) {
+    return NAMESPACE::make_format_args(vec[S]...);
 }
 
 template<std::size_t size>
-std::format_args&& unpack_vector(const std::vector<std::string>& vec) {
+NAMESPACE::format_args unpack_vector(const std::vector<std::string>& vec) {
     return unpack_vector(vec, std::make_index_sequence<size>());
 }
-
-#endif
 
 //' Simple Pass-Through Formatter to \code{fmt::format()}
 //'
@@ -36,52 +38,27 @@ std::format_args&& unpack_vector(const std::vector<std::string>& vec) {
 // [[Rcpp::export]]
 std::string formatter(const std::string s, std::vector<std::string> v) {
     size_t n = v.size();
-    #ifdef SPDLOG_USE_STD_FORMAT
     switch (n) {
-        case 0: return std::vformat(std::string_view(s), std::make_format_args());
-        case 1: return std::vformat(std::string_view(s), unpack_vector<1>(v));
-        case 2: return std::vformat(std::string_view(s), unpack_vector<2>(v));
-        case 3: return std::vformat(std::string_view(s), unpack_vector<3>(v));
-        case 4: return std::vformat(std::string_view(s), unpack_vector<4>(v));
-        case 5: return std::vformat(std::string_view(s), unpack_vector<5>(v));
-        case 6: return std::vformat(std::string_view(s), unpack_vector<6>(v));
-        case 7: return std::vformat(std::string_view(s), unpack_vector<7>(v));
-        case 8: return std::vformat(std::string_view(s), unpack_vector<8>(v));
-        case 9: return std::vformat(std::string_view(s), unpack_vector<9>(v));
-        case 10: return std::vformat(std::string_view(s), unpack_vector<10>(v));
-        case 11: return std::vformat(std::string_view(s), unpack_vector<11>(v));
-        case 12: return std::vformat(std::string_view(s), unpack_vector<12>(v));
-        case 13: return std::vformat(std::string_view(s), unpack_vector<13>(v));
-        case 14: return std::vformat(std::string_view(s), unpack_vector<14>(v));
-        case 15: return std::vformat(std::string_view(s), unpack_vector<15>(v));
+        case 0: return NAMESPACE::vformat(NAMESPACE::string_view(s), NAMESPACE::make_format_args());
+        case 1: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<1>(v));
+        case 2: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<2>(v));
+        case 3: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<3>(v));
+        case 4: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<4>(v));
+        case 5: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<5>(v));
+        case 6: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<6>(v));
+        case 7: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<7>(v));
+        case 8: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<8>(v));
+        case 9: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<9>(v));
+        case 10: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<10>(v));
+        case 11: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<11>(v));
+        case 12: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<12>(v));
+        case 13: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<13>(v));
+        case 14: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<14>(v));
+        case 15: return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<15>(v));
         default: {
             Rcpp::warning("Only up to fifteen arguments support for now.");
-            return std::vformat(std::string_view(s), unpack_vector<12>(v));
+            return NAMESPACE::vformat(NAMESPACE::string_view(s), unpack_vector<12>(v));
         }
     }
-    #else
-    switch (n) {
-        case 0: return fmt::format(s);
-        case 1: return fmt::format(s, std::string(v[0]));
-        case 2: return fmt::format(s, std::string(v[0]), std::string(v[1]));
-        case 3: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]));
-        case 4: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]));
-        case 5: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]));
-        case 6: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]));
-        case 7: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]));
-        case 8: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]));
-        case 9: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]));
-        case 10: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]));
-        case 11: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]), std::string(v[10]));
-        case 12: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]), std::string(v[10]), std::string(v[11]));
-        case 13: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]), std::string(v[10]), std::string(v[11]), std::string(v[12]));
-        case 14: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]), std::string(v[10]), std::string(v[11]), std::string(v[12]), std::string(v[13]));
-        case 15: return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]), std::string(v[10]), std::string(v[11]), std::string(v[12]), std::string(v[13]), std::string(v[14]));
-        default: {
-            Rcpp::warning("Only up to fifteen arguments support for now.");
-            return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]), std::string(v[10]), std::string(v[11]));
-        }
-    }
-#endif
 }
     

--- a/src/formatter.cpp
+++ b/src/formatter.cpp
@@ -3,6 +3,20 @@
 
 #include <RcppSpdlog>
 
+#ifdef SPDLOG_USE_STD_FORMAT
+
+template<std::size_t... S>
+std::format_args&& unpack_vector(const std::vector<std::string>& vec, std::index_sequence<S...>) {
+    return std::make_format_args(vec[S]...);
+}
+
+template<std::size_t size>
+std::format_args&& unpack_vector(const std::vector<std::string>& vec) {
+    return unpack_vector(vec, std::make_index_sequence<size>());
+}
+
+#endif
+
 //' Simple Pass-Through Formatter to \code{fmt::format()}
 //'
 //' The C-level interface of R does not make it easy to pass \code{...}  arguments.
@@ -20,21 +34,9 @@
 //' @return A single (formatted) string
 //' @seealso https://github.com/fmtlib/fmt
 // [[Rcpp::export]]
-
-#ifdef SPDLOG_USE_STD_FORMAT
-
-template<std::size_t... S>
-std::format_args&& unpack_vector(const std::vector<std::string>& vec, std::index_sequence<S...>) {
-    return std::make_format_args(vec[S]...);
-}
-
-template<std::size_t size>
-std::format_args&& unpack_vector(const std::vector<std::string>& vec) {
-    return unpack_vector(vec, std::make_index_sequence<size>());
-}
-
 std::string formatter(const std::string s, std::vector<std::string> v) {
     size_t n = v.size();
+    #ifdef SPDLOG_USE_STD_FORMAT
     switch (n) {
         case 0: return std::vformat(std::string_view(s), std::make_format_args());
         case 1: return std::vformat(std::string_view(s), unpack_vector<1>(v));
@@ -57,12 +59,7 @@ std::string formatter(const std::string s, std::vector<std::string> v) {
             return std::vformat(std::string_view(s), unpack_vector<12>(v));
         }
     }
-}
-
-#else
-
-std::string formatter(const std::string s, std::vector<std::string> v) {
-    size_t n = v.size();
+    #else
     switch (n) {
         case 0: return fmt::format(s);
         case 1: return fmt::format(s, std::string(v[0]));
@@ -85,6 +82,6 @@ std::string formatter(const std::string s, std::vector<std::string> v) {
             return fmt::format(s, std::string(v[0]), std::string(v[1]), std::string(v[2]), std::string(v[3]), std::string(v[4]), std::string(v[5]), std::string(v[6]), std::string(v[7]), std::string(v[8]), std::string(v[9]), std::string(v[10]), std::string(v[11]));
         }
     }
-}
-
 #endif
+}
+    

--- a/src/formatter.cpp
+++ b/src/formatter.cpp
@@ -20,6 +20,47 @@
 //' @return A single (formatted) string
 //' @seealso https://github.com/fmtlib/fmt
 // [[Rcpp::export]]
+
+#ifdef SPDLOG_USE_STD_FORMAT
+
+template<std::size_t... S>
+std::format_args&& unpack_vector(const std::vector<std::string>& vec, std::index_sequence<S...>) {
+    return std::make_format_args(vec[S]...);
+}
+
+template<std::size_t size>
+std::format_args&& unpack_vector(const std::vector<std::string>& vec) {
+    return unpack_vector(vec, std::make_index_sequence<size>());
+}
+
+std::string formatter(const std::string s, std::vector<std::string> v) {
+    size_t n = v.size();
+    switch (n) {
+        case 0: return std::vformat(std::string_view(s), std::make_format_args());
+        case 1: return std::vformat(std::string_view(s), unpack_vector<1>(v));
+        case 2: return std::vformat(std::string_view(s), unpack_vector<2>(v));
+        case 3: return std::vformat(std::string_view(s), unpack_vector<3>(v));
+        case 4: return std::vformat(std::string_view(s), unpack_vector<4>(v));
+        case 5: return std::vformat(std::string_view(s), unpack_vector<5>(v));
+        case 6: return std::vformat(std::string_view(s), unpack_vector<6>(v));
+        case 7: return std::vformat(std::string_view(s), unpack_vector<7>(v));
+        case 8: return std::vformat(std::string_view(s), unpack_vector<8>(v));
+        case 9: return std::vformat(std::string_view(s), unpack_vector<9>(v));
+        case 10: return std::vformat(std::string_view(s), unpack_vector<10>(v));
+        case 11: return std::vformat(std::string_view(s), unpack_vector<11>(v));
+        case 12: return std::vformat(std::string_view(s), unpack_vector<12>(v));
+        case 13: return std::vformat(std::string_view(s), unpack_vector<13>(v));
+        case 14: return std::vformat(std::string_view(s), unpack_vector<14>(v));
+        case 15: return std::vformat(std::string_view(s), unpack_vector<15>(v));
+        default: {
+            Rcpp::warning("Only up to fifteen arguments support for now.");
+            return std::vformat(std::string_view(s), unpack_vector<12>(v));
+        }
+    }
+}
+
+#else
+
 std::string formatter(const std::string s, std::vector<std::string> v) {
     size_t n = v.size();
     switch (n) {
@@ -45,3 +86,5 @@ std::string formatter(const std::string s, std::vector<std::string> v) {
         }
     }
 }
+
+#endif

